File: backend/JobPortalWebsite/JobPortal.API/Models/Application.cs
namespace JobPortal.API.Models;

public class Application
{
    public int Id { get; set; }
    public int JobId { get; set; }
    public int UserId { get; set; }
    public string? Status { get; set; }
    public string? ResumeUrl { get; set; }
}

File: frontend/src/app/models/application.ts
export interface JobApplication {
  id: number;
  jobId: number;
  userId: number;
  candidateName?: string | null;
  headline?: string | null;
  status: string | null;
  resumeUrl: string | null;
}

/** Candidate's application rows joined with job fields (`GET .../Applications/mine`). */
export interface MyApplication {
  id: number;
  jobId: number;
  jobTitle: string | null;
  company: string | null;
  location: string | null;
  status: string | null;
  resumeUrl: string | null;
}
